### PR TITLE
Documentation/139 classes modules

### PR DESCRIPTION
**Pull request by maudnals**
See https://github.com/KILTprotocol/sdk-js/issues/124 

To check how the API doc looks locally:  
1. Run `npm run build:docs`. This generates the doc at `sdk-js/docs/api`.   
2. Open any of the so-generated files in your browser (for example `sdk-js/docs/api/modules/uuid.html`). You can now navigate through the whole doc.

Build error at Step 1?   
=> Local hack (don't do this at home):   
Change`sdk-js/node_modules/@types/bs58/index.d.ts(8)` to make it look like:
```
import BaseConverter = require('base-x')
declare const base58: any
export = base58
```

### CODE REVIEW

File: src/attestation/Attestation.ts [hunk 1]
```diff
@@ -1,5 +1,15 @@
 /**
+ * Attestation provides the KILT ecosystem with handling [[Claim]] objects, which are then written on the [[Blockchain]].
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/attestation/Attestation.ts]: Changed to the certify claim object.

File: src/attestation/Attestation.ts [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+ * ***
+ *  Attestation issued by the Attester is sent to and stored with the claimer. We call these attested [[Claim]]s Credentials.
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/attestation/Attestation.ts]: Link to AttestedClaims.

File: src/attestedclaim/AttestedClaim.ts [hunk 3]
```diff
@@ -1,5 +1,15 @@
 /**
- * @module AttestationPresentation
+ * In KILT, the AttestedClaim is a **credential**, which a Claimer can store locally and share with Verifiers as they wish.
+ * ***
+ * Once a [[RequestForAttestation]] has been made, the [[Attestation]] can be built and the Attester submits it wrapped in an [[AttestedClaim]] object. This [[AttestedClaim]] also contains the original request for attestation.
+ * <br>
+ * [[RequestForAttestation]] also exposes a [[createPresentation]] method, that can be used by the claimer to hide some specific information from the verifier for more privacy.
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/attestedclaim/AttestedClaim.ts]: Remove the link from the RequestForAttestation.

File: src/blockchain/Blockchain.ts [hunk 4]
```diff
@@ -1,5 +1,13 @@
 /**
+ *  Blockchain bridges that connects the SDK and the KILT Blockchain.
+ *  ***
+ *  Communicates with the chain via [[listenToBlocks]], exposes the [[submitTx]] function that performs a transaction.
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/blockchain/Blockchain.ts]: I have updated the information.

File: src/blockchainApiConnection/BlockchainApiConnection.ts [hunk 5]
```diff
@@ -1,7 +1,14 @@
 /**
+ * Blockchain Api Connection enables the building and accessing of the KILT [[Blockchain]] connection.
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/blockchainApiConnection/BlockchainApiConnection.ts]: Updated the information.

File: src/ctype/CType.ts [hunk 6]
```diff
@@ -1,5 +1,15 @@
 /**
+ * CTypes are the way the KILT protocol creates a [[Claim]] type by enabling Attesters to use a certain [[Claim]] schema for creating specific credentials.
+ *  ***
+ *  * A CTYPE is a JSON-like description of a data structure.
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/ctype/CType.ts]: I have added the link

File: src/attestation/Attestation.ts [hunk 7]
```diff
@@ -1,5 +1,15 @@
...
+ *
+ *  Attestation objects are stored on a map within the [[Blockchain]], the claimHash used as a key and a tuple of [[CType]] hash. The Attester can revoke a [[Claim]].
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/attestation/Attestation.ts]: Added the tuple elements.

File: src/ctype/CType.ts [hunk 8]
```diff
@@ -1,5 +1,15 @@
 /**
+ * CTypes are the way the KILT protocol creates a [[Claim]] type by enabling Attesters to use a certain [[Claim]] schema for creating specific credentials.
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/ctype/CType.ts]: Adjusted the sentence for everyone, including Verifiers.

File: src/delegation/Delegation.ts [hunk 9]
```diff
@@ -1,5 +1,14 @@
 /**
+ * Delegations are the building blocks of top-down trust structures in KILT. An Attester can inherit trust through delegation from another attester ("top-down").
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/delegation/Delegation.ts]: Made some changes.

File: src/attestation/Attestation.ts [hunk 10]
```diff
@@ -3,7 +3,7 @@
  * ***
  *  Attestation issued by the Attester is sent to and stored with the claimer. We call these [[AttestedClaim]]s Credentials.
  *
- *  Attestation objects are stored on a map within the [[Blockchain]], the claimHash used as a key and a tuple of [[CType]] hash. The Attester can revoke a [[Claim]].
+ *  Attestation objects are stored on a map within the [[Blockchain]], the claimHash used as a key and a tuple of [[CType]] hash, account, delegation-id and revoked. The Attester can revoke a [[Claim]].
```
Comment (tjwelde) [inline src/attestation/Attestation.ts]: i would say "revoked flag"

File: src/attestedclaim/AttestedClaim.ts [hunk 11]
```diff
@@ -1,9 +1,9 @@
 /**
  * In KILT, the AttestedClaim is a **credential**, which a Claimer can store locally and share with Verifiers as they wish.
  * ***
- * Once a [[RequestForAttestation]] has been made, the [[Attestation]] can be built and the Attester submits it wrapped in an [[AttestedClaim]] object. This [[AttestedClaim]] also contains the original request for attestation.
+ * Once a RequestForAttestation has been made, the [[Attestation]] can be built and the Attester submits it wrapped in an [[AttestedClaim]] object. This [[AttestedClaim]] also contains the original request for attestation.
```
Comment (tjwelde) [inline src/attestedclaim/AttestedClaim.ts]: why did you remove the link?

File: src/ctype/CType.ts [hunk 12]
```diff
@@ -1,7 +1,7 @@
 /**
- * CTypes are the way the KILT protocol creates a [[Claim]] type by enabling Attesters to use a certain [[Claim]] schema for creating specific credentials.
+ * CTypes are the way the KILT protocol enables a Claimer or Attester or Verifier to create a [[Claim]] schema for creating specific credentials.
  *  ***
- *  * A CTYPE is a JSON-like description of a data structure.
+ *  * A CTYPE is a [Json-schema](http://json-schema.org/) description of a data structure.
```
Comment (tjwelde) [inline src/ctype/CType.ts]: sounds weird. maybe "A CTYPE is a description of the [[Claim]] data structure, based on [Json-schema](http://json-schema.org/)"

File: src/attestation/Attestation.ts [hunk 13]
```diff
@@ -1,7 +1,7 @@
 /**
- * Attestation provides the KILT ecosystem with handling [[Claim]] objects, which are then written on the [[Blockchain]].
+ * Attestation provides the KILT ecosystem with certifying [[Claim]] objects, which are then written on the [[Blockchain]].
```
Comment (tjwelde) [inline src/attestation/Attestation.ts]: Sounds a bit bumpy. Maybe "Attestations are used to certify [[Claim]] objects, ..." or similar

**Comment by tjwelde** [comment]
This PR was reset, to squash and merge it again in another PR
